test/Renci.SshNet.Tests/Common/AsyncSocketListener.cs

```diff
@@ -234,11 +234,7 @@ private void ReadCallback(IAsyncResult ar)
             try
             {
                 // Read data from the client socket.
-                bytesRead = handler.EndReceive(ar, out var errorCode);
-                if (errorCode != SocketError.Success)
-                {
-                    bytesRead = 0;
-                }
+                bytesRead = handler.EndReceive(ar);
```

Comment:
Despite being [documented](https://learn.microsoft.com/en-gb/dotnet/api/system.net.sockets.socket.endreceive?view=net-8.0#system-net-sockets-socket-endreceive(system-iasyncresult-system-net-sockets-socketerror@)) as throwing, `EndReceive(IAsyncResult, out SocketError)` actually [suppresses](https://github.com/dotnet/runtime/blob/5535e31a712343a63f5d7d796cd874e563e5ac14/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs#L2505) exceptions and translates them into the `out SocketError` parameter, at which point we would continue on and call `ConnectionDisconnected(true)`, giving another way to crash.